neurom/features/_impl.py

```diff
@@ -138,6 +137,24 @@ def section_path_distances(neurites, use_start_point=False, neurite_type=Neurite
     return iter_neurites(neurites, magic_iter, _ttc(neurite_type))
 
 
+def segment_radial_distances(neurites, neurite_type=NeuriteType.all):
+    '''Get an iterable containing section radial distances to origin of\
+       all neurites of a given type
+
+    Parameters:
+        origin: Point wrt which radial dirtance is calulated\
+                (default tree root)
+        use_start_point: if true, use the section's first point,\
+                         otherwise use the end-point (default False)
+        neurite_type: Type of neurites to be considered (default all)
+    '''
```

Comment:
Could you put the `i_segment_radial_dist` code in here? I really want to get away from these specialized iterator functions that don't anything useful. I want to remove these from morphtree, not add more.